noir-projects/noir-contracts/contracts/card_game_contract/src/cards.nr

```diff
@@ -116,12 +118,14 @@ impl Deck {
         let options = NoteGetterOptions::with_filter(filter_cards, cards);
         let maybe_notes = self.set.get_notes(options);
         let mut found_cards = [Option::none(); N];
+        let owner_npk_m_hash = get_npk_m_hash(self.set.context.private.unwrap(), owner);
+
         for i in 0..maybe_notes.len() {
             if maybe_notes[i].is_some() {
                 let card_note = CardNote::from_note(maybe_notes[i].unwrap_unchecked());
                 // Ensure the notes are actually owned by the owner (to prevent user from generating a valid proof while
                 // spending someone else's notes).
-                assert(card_note.note.owner.eq(owner));
+                assert(card_note.note.owner_npk_m_hash.eq(owner_npk_m_hash));
```

Comment:
Again, this is business logic and should not be in here, but be handled earlier up the stack. This is broken and should be fixed in #6312.